src/server/tran_server.cpp

```diff
@@ -358,21 +360,30 @@ tran_server::uses_remote_storage () const
 }
 
 void
-tran_server::push_request (tran_to_page_request reqid, std::string &&payload)
+tran_server::push_request_to (size_t idx, tran_to_page_request reqid, std::string &&payload)
 {
+  assert (idx < m_page_server_conn_vec.size());
   if (!is_page_server_connected ())
     {
       return;
     }
 
-  m_page_server_conn_vec[0]->push (reqid, std::move (payload));
+  m_page_server_conn_vec[idx]->push (reqid, std::move (payload));
+}
+
+void
+tran_server::push_request (tran_to_page_request reqid, std::string &&payload)
+{
+  // TODO push a request to the "main connection"
```

Comment:
this is already done :)
the first one is - by convention - the main connection

in a configuration with many PS's, the PTS's connection list should be:
- PTS1: PS1, PS2 (PS1 is the main connection)
- PTS2: PS1, PS2 ( -----``----)
- PTS3: PS2, PS1 (PS2 is the main connection)
- ...

This way some form of manual-static "load balancing" is obtained. This was the idea from the beginning.

Reply:
Thank you for the heads-up. I'm aware of this, but I think we will introduce the main connection explicitly in the code when handling failure of PS or a newly connected PS which has more priority. For example:

TS1: _PS1_ (not connected), PS2, PS3 
The vector[0] contains PS2. When the PS1 is connected, it should become the main connection and it would be in the vector[2]. Or we can shift existing ones.

I wanted to note that the `push_request` without a given index is supposed to the main connection.